src/cow-react/modules/wallet/web3-react/connectors/Ledger/provider.tsx

```diff
@@ -0,0 +1,154 @@
+import { TypedDataUtils, SignTypedDataVersion } from '@metamask/eth-sig-util'
+import { TypedDataDomain, TypedDataField } from '@ethersproject/abstract-signer'
+import type Transport from '@ledgerhq/hw-transport'
+import ProviderEngine from 'web3-provider-engine'
+import { LedgerOptions } from './connector'
+
+global.process = { browser: true } as any
+const RpcSubprovider = require('web3-provider-engine/subproviders/rpc')
+
+interface LedgerModules {
+  transportFactory: {
+    create(): Promise<Transport>
+  }
+  createLedgerSubprovider: (factory: () => Promise<Transport>, options: unknown) => any
+  LedgerEth: new (transport: Transport) => any
+}
+
+export interface EcdsaSignature {
+  r: string
+  s: string
+  v: number
+}
+
+// TODO: use current chain Id
+const currentChainId = 1
+const derivationPaths = ["44'/60'/x'/0/0", "44'/60'/0'/x"]
+
+function _generateDerivationPaths(context: { accountsOffset: number; accountsLength: number }): string[] {
+  const { accountsOffset, accountsLength } = context
+  const pathList: string[] = []
+
+  for (let i = accountsOffset; i < accountsOffset + accountsLength; i++) {
+    const x = Math.floor(i / derivationPaths.length)
+    const pathIndex = i - derivationPaths.length * x
+    const path = derivationPaths[pathIndex].replace('x', String(x))
+
+    pathList.push(path)
+  }
+
+  return pathList
+}
+
+function _getModule(): Promise<LedgerModules> {
+  return (
+    Promise.all([
+      // @ts-ignore
+      import('@ledgerhq/hw-transport-u2f'),
+      // @ts-ignore
+      import('@ledgerhq/hw-transport-webhid'),
+      // @ts-ignore
+      import('@ledgerhq/web3-subprovider'),
+      // @ts-ignore
+      import('@ledgerhq/hw-app-eth'),
+    ])
+      // eslint-disable-next-line @typescript-eslint/naming-convention
```

Comment:
Would be great to get rid off this and `@ts-ignore`. If it's possible

Reply:
Didn't find types for few of those 😞